llvm/lib/CodeGen/MachineInstr.cpp

```diff
@@ -2475,3 +2475,24 @@ MachineInstr::getFirst5RegLLTs() const {
       Reg2, getRegInfo()->getType(Reg2), Reg3, getRegInfo()->getType(Reg3),
       Reg4, getRegInfo()->getType(Reg4));
 }
+
+void MachineInstr::insert(mop_iterator It, ArrayRef<MachineOperand> Ops) {
+  assert(isVariadic() && "can only modify variadic instructions");
+  assert(It->getParent() == this && "iterator points to operand of other inst");
```

Comment:
Probably want to check this for every operand in the loop below?

Reply:
I don't follow; we need to check that the iterator for where in a given MachineInstr's list of operands to insert into points into that MachineInstr's operand list and not some other random/unrelated MachineIntr's operand list.

Reply:
I think I was thinking of something like this to make sure operands don't belong to some other instruction yet:
```
  for (MachineOperand* MOP : Ops) { assert(MOP->getParen() == nullptr); }
```
(which is a slightly different check from the iterator one you had there; sorry for confusion)

Reply:
Then again, the code ends up calling `addOperand()` and such an assert should rather be within that function then. So nothing to do here I guess.

Reply:
Also scratch this whole comment. It seems addOperand() and this insert() here actually copies the operands by value so no need for such a check at all.